textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/join/Join.java

```diff
@@ -120,14 +132,49 @@ public void open() throws Exception, DataFlowException {
      */
     @Override
     public ITuple getNextTuple() throws Exception {
+    	if (cursor == CLOSED) {
+            throw new DataFlowException(ErrorMessages.OPERATOR_NOT_OPENED);
+        }
+
         if (innerTupleList.isEmpty()) {
             return null;
         }
 
+        if (resultCursor >= limit + offset - 1 || limit == 0){
+            return null;
+        }
+
         ITuple nextTuple = null;
 
+        if (offset > 0) {
```

Comment:
Hi Sripad, please see "AbstractSingleInputOperator.java" and the function getNextTuple(). getNextTuple calls a function called "computeNextMatchingTuple", which returns the next matching tuple of the operator. Join should implement a similar function.